File: TriviaServer/TriviaServer/GetStatisticsResponse.cpp
#include "GetStatisticsResponse.h"

//------------constructor----------------
GetStatisticsResponse::GetStatisticsResponse(int _status, std::map<string,string> data)
{
	this->status = _status;
	this->messageCode = 71;
	this->statistics = data;
}

/*
The function will cast a signup response to JSON format
input: none
output: json format of this object
*/
json GetStatisticsResponse::castToJson() const
{
	return json{ {"status",this->status},
		{"gamesAmount",this->statistics.at("gamesAmount")},
		{"correctAnswers",this->statistics.at("correctAnswers")},
		{"incorrectAnswers",this->statistics.at("incorrectAnswers")},
		{"averageAnswerTime",this->statistics.at("averageAnswerTime")} };

}

File: TriviaWPFClient/TriviaClient/GetStatisticsResponse.cs
namespace TriviaClient
{
    class GetStatisticsResponse : Response
    {
        //#TODO (Omri): check how the ResponseDecoder builds this object, the stats attributes might be replaced with a string contains the data as an attribute.
        public const int MESSAGE_CODE = 71;
        public int status { get; set; }
        public int gamesAmount { get; set; }
        public int correctAnswers { get; set; }
        public int incorrectAnswers { get; set; }
        public double averageAnswerTime { get; set; }
        public GetStatisticsResponse() : base(MESSAGE_CODE) { }
    }
}
